Criado cadastro de atleta

LucaseNogueira opened the pull request:
Adicionado evento em no botão Cadastrar Atleta presente em "Competições"; Ao acessar a tela de Cadastro de atletas na competição, o sistema permite o cadastro tanto de um atleta já cadastrado no sistema quanto para criar um novo atleta e adiciona-lo na competição

LucaseNogueira committed:
Criado cadastro de atleta

Adicionado evento em no botão Cadastrar Atleta presente em "Competições";
Ao acessar a tela de Cadastro de atletas na competição, o sistema permite o cadastro tanto de um atleta já cadastrado no sistema quanto para criar um novo atleta e adiciona-lo na competição

LucaseNogueira committed:
Delete CadastrarTest.php

ghost commented:
![Failed](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/QualityGateBadge/failed.svg?sanitize=true)

- C Reliability Rating on New Code (is worse than A)
- 6.00% Coverage on New Code (is less than 80.00%)
- 16.93% Duplicated Lines (%) on New Code (is greater than 3.00%)

# Analysis Details
## 92 Issues
- ![Bug](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/bug.svg?sanitize=true) 2 Bugs
- ![Vulnerability](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 0 Vulnerabilities
- ![Code Smell](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 90 Code Smells

## Coverage and Duplications
- ![Coverage](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/CoverageChart/0.svg?sanitize=true) 6.00% Coverage (70.60% Estimated after merge)
- ![Duplications](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/Duplications/20.svg?sanitize=true) 16.93% Duplicated Code (5.00% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](https://sonarqube.mateux.dev/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=29)

luc527 commented on src/Competicoes/CompeticaoRepository.php:
```diff
@@ -40,6 +40,33 @@ public function todasAsCompeticoes(): array
+                   alterado_em
+              FROM competicao
+             WHERE id = $id
+          ORDER BY prazo DESC
```
Desnecessário o ORDER BY, esse select sempre vai trazer no máximo um resultado

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -0,0 +1,464 @@
+        $repo = new CompeticaoRepository(Connection::getInstance());
+        $competicao = $repo->buscarCompeticao($codigoCompeticao);
+        $competicao->setId($codigoCompeticao);
+        $competicao->setNome('Tomate Seco');
```
Desnecessários o `setId` e `setNome`

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -0,0 +1,464 @@
+        $repoAtleta = new AtletaCompeticaoRepository(Connection::getInstance());
+        $atletas = [];
+        foreach($repoAtleta->getAtletasForaCompeticaoViaNome($tecnico->id(), $codigoCompeticao) as $atleta){
+            $atletas[$atleta->nomeCompleto()] = $atleta->toJson();
```
Por que indexar pelo nome completo do atleta? Não poderia ser só `$atleta[] = $atleta->toJson();`?

luc527 commented on src/Tecnico/Atleta/AtletaCompeticao/AtletaCompeticaoRepository.php:
```diff
@@ -0,0 +1,164 @@
+        }
+    }
+
+    public function getAtletasForaCompeticaoViaNome(int $idTecnico, int $idCompeticao): array
```
Não entendi o "via nome" no nome do método, sugiro remover

luc527 commented on src/Tecnico/Atleta/AtletaCompeticao/AtletaCompeticaoRepository.php:
```diff
@@ -0,0 +1,164 @@
+            return $atletas;
+        }catch(Exception $e){
+            throw $e;
+            return [];
```
Remover o return, já tem um throw em cima

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -0,0 +1,464 @@
+        
+?>
+
+<main id="container-atletacompeticao" class="container">
```
O id não é utilizado em nenhum lugar, pode remover

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -0,0 +1,464 @@
+
+                <div class="card shadow mb-4">
+                    <div class="card-body pt-2">
+                        <ul class="nav mb-3 nav-underline">
```
O `nav-underline` não está disponível na versão 5.2.3 do boostrap, mas está na 5.3.0. Duas opções:
- Trocar a versão do bootstrap no head.html e script.html pela 5.3.0
  - `<link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-9ndCyUaIbzAi2FUVXJi0CjmCapSmO7SnpJef0486qhLnuZ2cdeRhO02iuK6FUUVM" crossorigin="anonymous">`
  - `<script src="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0/dist/js/bootstrap.bundle.min.js" integrity="sha384-geWF76RCwLtnZ8qwWowPQNguL3RmwHVBC9FhGdlKrxdiJJigb/j/68SIy3Te4Bkz" crossorigin="anonymous"></script>`
- Trocar o `nav-underline` por alguma das outras opções
  - `https://getbootstrap.com/docs/5.2/components/navs-tabs/#available-styles`
O que fazer fica a teu critério.

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -0,0 +1,464 @@
+<script src="https://ajax.googleapis.com/ajax/libs/jquery/2.1.1/jquery.min.js"></script>
+
+<!-- Importando o js do bootstrap -->
+<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
```
Remover essa tag <script>. O js do bootstrap já é importado na scripts.html, e, além disso, o script sendo importado aí é de uma versão antiga.
Caso ele seja necessário para alguma funcionalidade, usar no lugar a versão dessa funcionalidade do bootstrap 5.2.3 que é a versão que estamos usando.

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -0,0 +1,464 @@
+        background-color: rgba(var(--bs-light-rgb),var(--bs-bg-opacity)) !important;
+    }
+
+    .fake-link {
```
Remover, não é utilizado

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -0,0 +1,464 @@
+                        <div class="tab-content">
+                            <div id="selecionar_atleta" class="tab-pane show active">
+                                <div class="input-group mb-3 elementos-sem-atleta">
+                                    <input id="pesquisa-atleta" type="text" class="form-control" placeholder="Digite o nome do atleta...">
```
Atualmente o fluxo de pesquisa de atleta está meio estranho.
No protótipo, o usuário vai digitando o nome do atleta nesse campo e logo abaixo vai aparecendo os atletas com o nome buscado conforme o usuário digita.
Do jeito que ficou, ao digitar nesse campo não acontece nada. O usuário tem que clicar no ícone de pesquisar para abrir um modal onde realmente acontece a pesquisa. Porém, mesmo digitando o nome do atleta, não acontece nada, e ele tem que clicar no ícone de pesquisa novamente. Aí finalmente que ele consegue selecionar os atletas.
Poderia ser feito o seguinte:
- Remover a barra de pesquisa que aparece direto na tela, deixar só um botão "pesquisar atleta".
- Ao clicar nesse botão abre o modal.
- Nesse modal, fazer com que os atletas sejam filtrados conforme o usuário digita (usar o debounce do utils.js pra não iterar sobre todos os atletas *a cada tecla digitada*, mas esperar o usuário terminar de digitar).
- Caso nenhum atleta tenha nome parecido com o que foi digitado, mostrar um alerta de "nenhuma tleta encontrado"

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -0,0 +1,464 @@
+
+</main>
+
+<div id="consulta-atleta" class="modal fade" tabindex="-1" role="dialog" aria-labelledby="myLargeModalLabel" aria-hidden="true">
```
Falta fechar essa div, acabou que o footer ficou dentro dela:

![image](https://github.com/MateuxDotDev/badminton-manager/assets/48414967/e4746771-5f9d-49a5-8185-5a3c68cd2b9c)

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -0,0 +1,464 @@
+    function reloadModalListaAtleta(nomeAtleta){
+        var atletasFiltrados = getAtletasFiltradosModal(nomeAtleta);
+        var listaAtletaElement = document.getElementById("lista_atleta_modal");
+        limpar(listaAtletaElement);
```
Já tem uma função `esvaziar` no `utils.js`.
Trocar os usos da função `limpar` por `esvaziar` e remover a declaração da função `limpar`.

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -0,0 +1,464 @@
+            </div>
+        </div>
+
+        <div class="modal-footer">
```
Remover o modal-footer

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -0,0 +1,464 @@
+            roundedCircle.classList.add("rounded-circle");
+            roundedCircle.style.height = "60px";
+            roundedCircle.style.width = "60px";
+            roundedCircle.style.backgroundColor = "crimson";
```
Pode remover a cor de fundo, coloquei no protótipo só pra distinguir a foto de perfil
Remover os outros usos também (ctrl+f "crimson")

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -0,0 +1,464 @@
+            roundedCircle.style.backgroundColor = "crimson";
+            if(atleta.foto != ''){
+                let imgRoundedCircle = document.createElement("img");
+                imgRoundedCircle.src = atleta.foto;
```
A src tem que ser
```js
`/assets/images/profile/${atleta.foto}`
```

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -0,0 +1,464 @@
+    const atletas = Object.values(<?= json_encode($atletas) ?>);
+    var atletaSelecionado;
+
+    form.addEventListener('submit', (event)=>{
```
Falta validar:
- Se o "selecionar atleta cadastrado" estiver selecionado, ele tem que selecionar algum atleta
- Se o "cadastrar novo atleta" estiver selecionado, ele tem que validar que o atleta tem nome e data de nascimento
- Pelo menos uma categoria tem que ter sido selecionada
- Pelo menos um tipo de dupla tem que ter sido selecionado

luc527 commented on public/atletacompeticao/acao.php:
```diff
@@ -0,0 +1,273 @@
+
+function cadastrarAtletaCompeticao($pdo, Atleta $atleta): Response
+{
+    if(!$atleta->id()){
```
A função vai ficar mais simples se sempre receber um atleta pronto, de forma que o lugar onde essa função é chamada que é responsável por fazer o getAtletaById se necessário (a realizarCadastroAtletaSelecionado)

luc527 commented on public/atletacompeticao/acao.php:
```diff
@@ -0,0 +1,273 @@
+    return $repo->getAtletaViaId($_POST["atleta"]);
+}
+
+function validaCategoriaAtleta(Categoria $categoria, Atleta $atleta): bool
```
Já existe um método na clase `Categoria` que faz isso, o `podeParticipar`. Trocar o uso da validaCategoriaAtleta por uma chamada a esse método e remover a função.

luc527 commented on public/atletacompeticao/acao.php:
```diff
@@ -0,0 +1,273 @@
+        throw new ValidatorException('Data de nascimento inválida');
+    }
+
+    $dataNascimento->setTime(0, 0);
```
A $dataNascimento é do tipo `DateTimeInterface` que não tem o método `setTime` disponível:
![image](https://github.com/MateuxDotDev/badminton-manager/assets/48414967/e9220783-e94b-4d1c-bb68-d5fe8c76e30c)
Acho que dá pra remover o `setTime`, até porque só salvamos a data no postgres, não as horas/minutos/segundos.

luc527 commented on src/Tecnico/Atleta/AtletaCompeticao/AtletaCompeticaoCategoriaRepository.php:
```diff
@@ -0,0 +1,72 @@
+                INSERT INTO atleta_competicao_categoria (
+                    atleta_id,
+                    competicao_id,
+                    categoria
```
Trocar por categoria_id

luc527 replied:
Após aquela alteração que fiz na tabela `categoria`, as outras não referenciam mais a categoria pela descrição, mas pelo id, então todas as colunas que eram `categoria text references categoria(descricao)` viraram `categoria_id bigint references categoria(id)`, e essa coluna da `atleta_competicao_categoria` foi uma delas

luc527 commented on src/Tecnico/Atleta/AtletaCompeticao/AtletaCompeticaoCategoriaRepository.php:
```diff
@@ -0,0 +1,72 @@
+            $stmt->execute([
+                'atleta_id' => $aca->atletaCompeticao()->atleta()->id(),
+                'competicao_id' => $aca->atletaCompeticao()->competicao()->id(),
+                'categoria' => $aca->categoria()->descricao()
```
Trocar por `$aca->categoria()->id()`

luc527 commented on src/Tecnico/Atleta/AtletaRepository.php:
```diff
@@ -107,4 +105,54 @@ public function getViaTecnico(int $tecnicoId): array
+    }
+
+    public function defineTransaction(bool $define){
+        $this->defineTransaction = $define;
```
Não existe atributo `defineTransaction`, criar lá no topo pra poder usar

MateuxLucax commented:
@LucaseNogueira Antes da revisão, pode já ir resolvendo as coisas citadas pelo comentário do SonarQube. O acesso pro SonarQube enviei pelo Discord.

luc527 commented on public/atletacompeticao/acao.php:
```diff
@@ -0,0 +1,273 @@
+        $pdo->beginTransaction();
+        $retorno = cadastrarAtletaCompeticao($pdo, new Atleta());
+
+        $pdo->commit();
```
Pode ser que um `rollback` já tenha sido feito quando o código chega aqui, causando um erro de "There is no active transaction". De fato, sempre que o `$retorno` é uma `Response` com erro (por exemplo quando o atleta não pode jogar na categoria selecionada pela idade), é feito antes um `rollback`:
![image](https://github.com/MateuxDotDev/badminton-manager/assets/48414967/ffcff2ca-c4e8-48b5-b6a3-6fed3b39e256)
A solução mais direta seria verificar se o pdo está dentro de uma transaction antes de fazer o commit, mas acho que isso não lida diretamente com a raíz do problema. O verdadeiro problema é que deveríamos fazer todas as validações (verificar se vieram categorias e duplas na request, verificar se o atleta pode jogar nas categorias que selecionou, verificar se atleta já não está cadastrado na competição etc. etc.) **antes** de iniciar a transaction. Aí nem teríamos que lidar com esse problema de rollback seguido de commit.

luc527 commented on public/atletacompeticao/acao.php:
```diff
@@ -0,0 +1,273 @@
+    };
+}
+
+function realizarCadastro($pdo) : Response
```
Tem categorias e sexo de dupla que não são registradas ao fazer o cadastro:
Selecionei três categorias e os dois tipos de dupla:
![image](https://github.com/MateuxDotDev/badminton-manager/assets/48414967/f45ae05d-2882-4bd1-b155-77b069c7288a)

Mas só registrou um de cada:
![image](https://github.com/MateuxDotDev/badminton-manager/assets/48414967/833936c3-ff72-4072-a02f-22705127789b)

LucaseNogueira committed:
Correção de bugs e aplicação de boas práticas

1. # CompeticaoRepository

	1. ### buscarCompeticao()
		1. Retirado order by desnecessário;

---

2. # atletacompeticao\index.php:
	1. Retirado o mock de competição;
	2. Retirado index "nome_atleta" no array de atletas apresentado nos campos de busca. Esse index não é utilizado em numa etapa do código fonte.
	3. Agora as imagens dos atletas selecionados aparecem na modal e na div de atleta selecionado;

---

3. # AtletaCompeticaoRepository

	1. ### getAtletasForaCompeticaoViaNome():
		1. Alterado o nome da função. Agora ela passa a se chamar "getAtletasForaCompeticao()".

	2. ### getAtletaCompeticao():
		1. Removido "return" presente após o "throw" no "catch" do método (recomendação do revisor e do sonarqube).

---

4. # Banco de dados local:

	1. Realizado a limpeza da base local do programador. Existiam relacionamentos que estavam inconformes com a main.
---

5. # AtletaCompeticaoCategoriaRepository:

	1. cadastrarAtletaCompeticaoCategoria():
		1. Alterado o nome do parametro "$aca" para "$acc";
		2. Alterado "categoria" por "categoria_id";
		3. Alterado "$acc->categoria()->descricao()" para "$acc->categoria()->id()"

---

6. # acao:

	1. Corrigido bug de rollback. Agora o sistema valida os dados que serão cadastrados antes de realizar a execução da transation

ghost commented:
![Failed](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/QualityGateBadge/failed.svg?sanitize=true)

- C Reliability Rating on New Code (is worse than A)
- 7.60% Coverage on New Code (is less than 80.00%)
- 16.81% Duplicated Lines (%) on New Code (is greater than 3.00%)

# Analysis Details
## 92 Issues
- ![Bug](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/bug.svg?sanitize=true) 1 Bug
- ![Vulnerability](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 0 Vulnerabilities
- ![Code Smell](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 91 Code Smells

## Coverage and Duplications
- ![Coverage](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/CoverageChart/0.svg?sanitize=true) 7.60% Coverage (71.10% Estimated after merge)
- ![Duplications](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/Duplications/20.svg?sanitize=true) 16.81% Duplicated Code (5.00% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](https://sonarqube.mateux.dev/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=29)

LucaseNogueira committed:
Retirado return desnecessário

Corrigindo bug acusado pelo SonarQube

ghost commented:
![Failed](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/QualityGateBadge/failed.svg?sanitize=true)

- 7.60% Coverage on New Code (is less than 80.00%)
- 16.84% Duplicated Lines (%) on New Code (is greater than 3.00%)

# Analysis Details
## 92 Issues
- ![Bug](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/bug.svg?sanitize=true) 0 Bugs
- ![Vulnerability](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 0 Vulnerabilities
- ![Code Smell](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 92 Code Smells

## Coverage and Duplications
- ![Coverage](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/CoverageChart/0.svg?sanitize=true) 7.60% Coverage (71.20% Estimated after merge)
- ![Duplications](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/Duplications/20.svg?sanitize=true) 16.84% Duplicated Code (5.00% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](https://sonarqube.mateux.dev/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=29)

MateuxLucax commented:
@LucaseNogueira Podes utilizar o SonarLint (consegue conectar ele na instancia do SonarQube nossa tambem) e com isso ele marca as coisas que precisa corrigir, principalmente o Code Smell. Isso faz parte da disciplina de Qualidade, então precisa estar 100% para ser aprovado o merge. E a parte de testes e duplicidade de código também.

LucaseNogueira committed:
Atualizações no Front-End e Boas Práticas

1 # atletacompeticao\Index.php:
	1. Retirado "id" presente na tag main;
	2. Substituido "nav-underline" por "nav-tabs";
	3. Retirado do css a classe ".fake-link";
	4. Retirado scripts ".js" do bootstrap presente no fim do arquivo;
	5. Fechado a div da modal.
	6. Adicionado validação quando a opção de cadastro for "Selecionar Atleta Cadastrado" o sistema emitir uma mensagem quando não foi informado nenhum atleta;
	7. Retirado o campo de pesquisa da tela de cadastro de atleta na competição.
	8. Alterado o estilo do botão de pesquisar. Removido a lupa presente no campo e adicionado o texto "Consultar Atletas";
	9. Alterado o campo de pesquisa da modal para retorar os atletas conforme o usuário vai digitando

2 # atletacompeticao\acao.php:
	1. Agora o sistema realiza as validações primárias antes de abrir uma conexão com o banco;
	2. Agora é passado um array de dados validados;

ghost commented:
![Failed](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/QualityGateBadge/failed.svg?sanitize=true)

- 7.60% Coverage on New Code (is less than 80.00%)
- 16.84% Duplicated Lines (%) on New Code (is greater than 3.00%)

# Analysis Details
## 92 Issues
- ![Bug](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/bug.svg?sanitize=true) 0 Bugs
- ![Vulnerability](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 0 Vulnerabilities
- ![Code Smell](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 92 Code Smells

## Coverage and Duplications
- ![Coverage](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/CoverageChart/0.svg?sanitize=true) 7.60% Coverage (71.20% Estimated after merge)
- ![Duplications](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/Duplications/20.svg?sanitize=true) 16.84% Duplicated Code (5.00% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](https://sonarqube.mateux.dev/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=29)

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -111,15 +107,14 @@
-                                    <button id="btn-pesquisar" class="input-group-text" type="button">
-                                        <i class="bi bi-search"></i>
+                                    <button id="btn-pesquisar" class="btn btn-outline-primary" type="button">
+                                        Consultar Atleta
```
Ainda pode deixar o ícone de lupa junto do botão (estamos usando os ícones do https://icons.getbootstrap.com/). Também acho que fica melhor como "buscar atleta" do que "consultar atleta".

```html
<button class="btn btn-outline-primary" type="button">
  <i class="bi bi-search"></i>
  Buscar atleta
</button>
```

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -111,15 +107,14 @@
                         <div class="tab-content">
                             <div id="selecionar_atleta" class="tab-pane show active">
                                 <div class="input-group mb-3 elementos-sem-atleta">
```
O mb-3 cria um espaço branco estranho quando não tem nenhum atleta selecionado:
![image](https://github.com/MateuxDotDev/badminton-manager/assets/48414967/38a96e56-8fd5-4769-8f88-059383c13be1)
Dá pra corrigir tirando o mb-3 do botão e colocando um mt-3 (margin-top) no elemento do atleta, aí quando esse elemento é escondido a margem some também

luc527 commented on public/atletacompeticao/index.php:
```diff
@@ -95,7 +91,7 @@
 
```
Margem faltando entre esses campos (sexo e observações). Provavelmente falta fechar essa `div` no lugar certo.

![image](https://github.com/MateuxDotDev/badminton-manager/assets/48414967/c6b78681-f86a-4353-9924-58a2b0a29046)

luc527 commented on public/atletacompeticao/acao.php:
```diff
@@ -181,21 +185,30 @@ function cadastrarAtletaCompeticaoDupla(PDO $pdo, AtletaCompeticao $atletaCompet
         foreach($categorias as $chave => $categoria){
-            if(!validaCategoriaAtleta($categoria, $atleta)){
+            if(!$categoria->podeParticipar($atleta->dataNascimento(), $competicao->prazo())){
                 throw new ValidatorException('A categoria ' . $categoria->descricao() . ' selecionada se torna invalida com relação a idade do(a) atleta');
```
Acho que o texto poderia ser um pouco mais simples, tipo "O atleta não tem uma idade adequada para jogar na categoria Sub 13"

luc527 commented:
Tem ainda esse erro (variável `$categoria` não existe no contexto, provavelmente era pra ser `$categoriaCompeticao`) no `public\atletacompeticao\acao.php`

![image](https://github.com/MateuxDotDev/badminton-manager/assets/48414967/45653aed-44f7-44f2-a209-c378d932c5f1)

LucaseNogueira committed:
Aplicado Melhorias Fornt-End e Correção de bug

# Front-End:
1. Retornado icone de lupa no botão de "Consultar Atleta";
2. Alterado no nome do botão "Consultar Atleta" para "Buscar Atleta"
3. Adicionado margem na div de atleta selecionado e retirado margem-inferior da div superior
4. Agora "sexo" e "Data de Nascimento" são componentes de uma div "row mb-3";
5. Alterado o conteúdo da mensagem apresentada quando o atleta não pode ser cadastrado com a categoria informada.

# Acao:
1. Corrigido bug relacionado a variavel $categoria não existir no contexto aplicado

ghost commented:
![Failed](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/QualityGateBadge/failed.svg?sanitize=true)

- 7.60% Coverage on New Code (is less than 80.00%)
- 16.84% Duplicated Lines (%) on New Code (is greater than 3.00%)

# Analysis Details
## 92 Issues
- ![Bug](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/bug.svg?sanitize=true) 0 Bugs
- ![Vulnerability](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 0 Vulnerabilities
- ![Code Smell](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 92 Code Smells

## Coverage and Duplications
- ![Coverage](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/CoverageChart/0.svg?sanitize=true) 7.60% Coverage (71.20% Estimated after merge)
- ![Duplications](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/Duplications/20.svg?sanitize=true) 16.84% Duplicated Code (5.00% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](https://sonarqube.mateux.dev/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=29)

LucaseNogueira committed:
Aplicado Boas Práticas - Sonarqube

ghost commented:
![Failed](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/QualityGateBadge/failed.svg?sanitize=true)

- 7.50% Coverage on New Code (is less than 80.00%)
- 18.00% Duplicated Lines (%) on New Code (is greater than 3.00%)

# Analysis Details
## 2 Issues
- ![Bug](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/bug.svg?sanitize=true) 0 Bugs
- ![Vulnerability](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 0 Vulnerabilities
- ![Code Smell](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 2 Code Smells

## Coverage and Duplications
- ![Coverage](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/CoverageChart/0.svg?sanitize=true) 7.50% Coverage (71.00% Estimated after merge)
- ![Duplications](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/Duplications/20.svg?sanitize=true) 18.00% Duplicated Code (5.50% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](https://sonarqube.mateux.dev/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=29)

LucaseNogueira committed:
Correção - logic catch clause (Sonarqube)

ghost commented:
![Failed](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/QualityGateBadge/failed.svg?sanitize=true)

- 7.50% Coverage on New Code (is less than 80.00%)
- 17.94% Duplicated Lines (%) on New Code (is greater than 3.00%)

# Analysis Details
## 0 Issues
- ![Bug](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/bug.svg?sanitize=true) 0 Bugs
- ![Vulnerability](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/vulnerability.svg?sanitize=true) 0 Vulnerabilities
- ![Code Smell](https://sonarqube.mateux.dev/static/communityBranchPlugin/common/code_smell.svg?sanitize=true) 0 Code Smells

## Coverage and Duplications
- ![Coverage](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/CoverageChart/0.svg?sanitize=true) 7.50% Coverage (70.80% Estimated after merge)
- ![Duplications](https://sonarqube.mateux.dev/static/communityBranchPlugin/checks/Duplications/20.svg?sanitize=true) 17.94% Duplicated Code (5.50% Estimated after merge)

**Project ID:** MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7

[View in SonarQube](https://sonarqube.mateux.dev/dashboard?id=MateuxDotDev_badminton-manager_AYfNu-knGxuymmoUfIu7&pullRequest=29)

luc527 reviewed (approved):
Ainda ficou faltando coverage no sonarqube, mas isso pode ser corrigido numa outra branch, senão o desenvolvimento não vai pra frente
